### DISCUSSION TITLE
Internalizing Blueretro on various systems, What is the best ESP32 option?

### DISCUSSION THREAD

**Discussion by vsilvalopes** [darthcloud/BlueRetro#367] [Q&A] [answered]
Hi everyone!
After getting success on Blueretro with PS1 and PS2, I'm thinking about going with all internal installs in my systems.

My idea is to internalize Blueretro, and with a mechanical switch cut the power of the module, so when I want/need to use wired controllers, it would work.

Some systems like Genesis, NES, SNES, PS1, have plenty of unused space inside.
But systems like PS2, PC-Engine, Gamecube, Dreamcast and N64 doesn't have that much space.

I'm thinking going this route, because I can have the systems without any dongle and with the option to use bluetooth controllers.

So, my question is, this module is compatible with Blueretro? 

https://pt.aliexpress.com/item/1005003309510459.html?spm=a2g0o.detail.1000023.7.6d964eeemlbP8u&gatewayAdapt=glo2bra

On the page says "ESP32-WROOM-32" 
I can't see a micro-USB connection, so to flash BlueRetro on it, I would probably need to solder a USB Cable on it right?

If this one is not compatible, can you guys point me to a model that is?
I'm planing to internalize on those systems:

- PS1
- PS2
- Genesis
- Saturn
- Dreamcast
- NES
- SNES
- N64
- GameCube
- PC-Engine

Thanks in advance!

**Comment by maniaco007** [comment]
Great topic!

I plan to do the internalization on my consoles as well.

This item you mentioned, looks like the same ones used inside some dongles for sale in the market! I'm not sure how the firmware is included in it...

On the question of including an on and off switch, would that be necessary? If I want to play with the original controls, with Blueretro installed internally, does it need to be turned off?

**Reply by vsilvalopes** [reply]
Hi!

I think it would be necessary to turn off the ESP module.

As it can bring some unwanted voltage in the wrong spot with the controller connected.

But as Darthcloud said, I will wait for the internal install docs. :)

**Comment by darthcloud** [comment, marked as answer]
Official support for internal install FW and docs are couple month away. Wait for that.

**Reply by vsilvalopes** [reply]
Wow! Thanks for this! 

I will just wait! :)

I was going first for the cables route, but putting the 74AHCT inside the DB25 connector will be a pain using just wires.

Also, the cables for the consoles that I have in AliExpress are way too expensive. 

So for me in Brazil is more cheap to buy 8 ESP modules and internalize than buying the cables and going the DB25 route.

Thanks for the update! 
Love the project!

**Comment by maniaco007** [comment]
Opa, sou do Brasil também!
Tenho o ps1 com o Xstation e com a Fonte Psu da Bitfunx.
Vou ligar o esp atraves da fonte usando um regulador 7805

**Reply by vsilvalopes** [reply]
Opa! 
Também tenho um PS1 com Xstation! Coisa fina né? :) 

Bacana encontrar um colega BR aqui!

Abraço!

**Comment by maniaco007** [comment]
Trazendo uma atualização no meu projeto, fiz a instalação interna no meu PS1 Fat, segue uma foto de como acomodei o ESP32. Utilizei cabos de arduino. Liguei ele direto na PSU , como a placa devkit possui um regulador nela, utilizei o pino de 8v.
os próximos passos agora serão a pintura das placas metálicas, um botão de liga desliga o esp32 ai em conjunto com a porta paralela, assim nao preciso modificar em nada a carcaça, e um cooler interno pra ajudar na dissipação de calor.

![IMG_20220904_191723](https://user-images.githubusercontent.com/82119174/188335731-e3d5605e-b562-4044-b1ac-7938152b6f00.jpg)

**Comment by leonrrafael** [comment]
> Hi everyone! After getting success on Blueretro with PS1 and PS2, I'm thinking about going with all internal installs in my systems.
> 
> My idea is to internalize Blueretro, and with a mechanical switch cut the power of the module, so when I want/need to use wired controllers, it would work.
> 
> Some systems like Genesis, NES, SNES, PS1, have plenty of unused space inside. But systems like PS2, PC-Engine, Gamecube, Dreamcast and N64 doesn't have that much space.
> 
> I'm thinking going this route, because I can have the systems without any dongle and with the option to use bluetooth controllers.
> 
> So, my question is, this module is compatible with Blueretro?
> 
> https://pt.aliexpress.com/item/1005003309510459.html?spm=a2g0o.detail.1000023.7.6d964eeemlbP8u&gatewayAdapt=glo2bra
> 
> On the page says "ESP32-WROOM-32" I can't see a micro-USB connection, so to flash BlueRetro on it, I would probably need to solder a USB Cable on it right?
> 
> If this one is not compatible, can you guys point me to a model that is? I'm planing to internalize on those systems:
> 
> * PS1
> * PS2
> * Genesis
> * Saturn
> * Dreamcast
> * NES
> * SNES
> * N64
> * GameCube
> * PC-Engine
> 
> Thanks in advance!

Estou tentando fazer o mesmo com consoles pouco difundido, tipo sega cd-x. Você teve sorte com algum? Consegui fazer bem fácil no Dreamcast.
